backend/routes/addUnknowns.js

```diff
@@ -37,41 +36,37 @@ router.post('/addUnknowns', async (req, res) => {
 
     // Update each document based on user information
     const batch = admin.firestore().batch();
-    activePlayersSnapshot.docs.forEach(doc => {
+    activePlayersSnapshot.forEach(doc => {
       const courtNumber = doc.data().courtNumber;
-      const isOccupied = occupiedCourts.includes(courtNumber);
-      const isEmptyCourt = doc.data().firebaseUID.startsWith('Empty');
+      const isPhysicallyOccupied = occupiedCourts.includes(courtNumber);
+      const isVirtuallyEmpty = doc.data().firebaseUID.startsWith('Empty');
 
-      // If court is occupied, check if it's shown as empty
-      if (isOccupied) {
-        if (isEmptyCourt) {
-          const newName = `Unknown${courtNumber}`;
-          const timestamp = timestamps[lastTimestampIdx];
-          lastTimestampIdx += 1;
+      // If court is physically occupied, check if it's virtually shown as empty
+      if (isPhysicallyOccupied && isVirtuallyEmpty) {
+        const newName = `Unknown${courtNumber}`;
+        const timestamp = timestamps[lastTimestampIdx];
+        lastTimestampIdx += 1;
 
-          const updateObj = {
-            playerWaiting: false,
-            firebaseUID: newName,
-            nickname: newName,
-            timeStartedPlay: timestamp,
-          };
+        const updateObj = {
+          playerWaiting: false,
+          firebaseUID: newName,
+          nickname: newName,
+          timeStartedPlay: timestamp,
+        };
 
-          batch.set(doc.ref, updateObj, { merge: true });
-        }
+        batch.set(doc.ref, updateObj, { merge: true });
 
-      // If court is empty, check if it's shown as occupied
-      } else {
-        if (!isEmptyCourt) {
-          const newName = `Empty${courtNumber}`;
+      // If court is physically empty, check if it's virtually shown as occupied
+      } else if (!isPhysicallyOccupied && !isVirtuallyEmpty) {
+        const newName = `Empty${courtNumber}`;
 
-          const updateObj = {
-            playerWaiting: false,
-            firebaseUID: newName,
-            nickname: newName,
-          };
+        const updateObj = {
+          playerWaiting: false,
+          firebaseUID: newName,
+          nickname: newName,
+        };
```

Comment:
Refactoring this if statement is a great addition! However, I believe there are 2 other possibilities when `!isPhysicallyOccupied && isVirtuallyEmpty` (No one on the court both physically and virtually) and `isPhysicallyOccupied && !isVirtuallyEmpty` (Someone is on the court both physically and virtually). In both cases, it wouldn't make sense to call this endpoint, but it's still possible to (if there are no restrictions in the frontend). Should these cases also be handled and return an error?

Reply:
Hey Leandro, good observation. Regarding these 2 cases, this means that our database has the up to date information on those courts. Our main focus is the other 2 cases currently identified as that means there is a mismatch.